### PR TITLE
Add support for Turkish translations

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
app-project
lib-classifier
lib-react-components

## Linked Issue and/or Talk Post
Follow up to added `tr` support in PFE: https://github.com/zooniverse/Panoptes-Front-End/pull/6160

## Describe your changes
- Added converted dictionaries to all three packages (via [pfe-fem-translation-keys](https://github.com/zooniverse/pfe-fem-translation-keys)).
- Added `tr` to i18n configuration files
- When locale is set to Turkish and text is transformed to uppercase, the Turkish "i" is applied whether the uppercase word is actually Turkish or not. As a result, I had to override the username label in SignedInUserNavigation as well as the username is TeamMember to always have `lang="en"` because a username should never display characters other than the latin alphabet.
- I updated lib-classifier's /dev localeMenu to match app-project's localeMenu.

## How to Review
- Make sure to yarn bootstrap
- Visit a project that uses Survey Task because that particular page will pull keys from all 5 translation dictionaries of the currently selected locale. Examples:
https://local.zooniverse.org:3000/projects/tr/wildcam/wildcam-darien/classify/workflow/3033?env=production
https://local.zooniverse.org:3000/projects/tr/markb-panoptes/survey-task-test-project/classify/workflow/3611?env=staging

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [ ] The PR creator has listed user actions to use when testing the new feature

### CODE REVIEW

File: packages/lib-react-components/src/ZooHeader/components/NarrowMenuNavListItem/NarrowMenuNavListItem.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 import SpacedText from '../../../SpacedText'
 
-export default function NarrowMenuNavListItem ({ color, text }) {
+export default function NarrowMenuNavListItem ({ color, text, overrideLang }) {
+  const { i18n } = useTranslation()
   return (
     <SpacedText
       color={color}
-      weight='bold'
+      lang={overrideLang ? 'en' : i18n.language}
```
Comment (eatyourgreens) [inline packages/lib-react-components/src/ZooHeader/components/NarrowMenuNavListItem/NarrowMenuNavListItem.js:12]: Is this setting the navigation menu to English, even if the translated text is in another language?

File: packages/lib-react-components/src/ZooHeader/components/UserMenu/UserMenu.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ export default function UserMenu ({ signOut, user }) {
   // Support staging urls...
   const host = getHost()
 
-  const userDisplayName = <NarrowMenuNavListItem color='#B2B2B2' text={user.display_name} />
+  const userDisplayName = <NarrowMenuNavListItem color='#B2B2B2' text={user.display_name} overrideLang />
```
Comment (eatyourgreens) [inline packages/lib-react-components/src/ZooHeader/components/UserMenu/UserMenu.js:23]: Maybe allow `lang` as an optional prop here.
```suggestion
  const userDisplayName = <NarrowMenuNavListItem color='#B2B2B2' text={user.display_name} lang='en' />
```

File: packages/lib-react-components/src/ZooHeader/components/NarrowMenuNavListItem/NarrowMenuNavListItem.js (hunk 1 above)
Comment (eatyourgreens) [inline packages/lib-react-components/src/ZooHeader/components/NarrowMenuNavListItem/NarrowMenuNavListItem.js:12]: Maybe add `lang` as an optional prop. That might be easier to read and understand.
```jsx
export default function NarrowMenuNavListItem({ color, text, lang }) {
…
  return (
    <SpacedText
      color={color}
      lang={lang}
      …
    />
```

Comment (goplayoutside3) [reply packages/lib-react-components/src/ZooHeader/components/NarrowMenuNavListItem/NarrowMenuNavListItem.js:12]: It is setting the label of the dropdown button who's text is username.

Comment (goplayoutside3) [reply packages/lib-react-components/src/ZooHeader/components/NarrowMenuNavListItem/NarrowMenuNavListItem.js:12]: Can you clarify why `lang` should be considered optional here? If the `lang` prop is undefined, won't the `lang` attribute be set to "undefined"? I think that could be confusing in the html structure.

Comment (eatyourgreens) [reply packages/lib-react-components/src/ZooHeader/components/NarrowMenuNavListItem/NarrowMenuNavListItem.js:12]: Undefined values aren’t rendered by React, so `lang` would be ignored here by default.

Comment (goplayoutside3) [reply packages/lib-react-components/src/ZooHeader/components/NarrowMenuNavListItem/NarrowMenuNavListItem.js:12]: I see! Yes that does make more sense for readability.

**Review by eatyourgreens — APPROVED** [review]
Looks good to me. I'm not sure about the implications of forcing usernames to be announced as English words in screen readers, but it seems like a reasonably harmless change.

**Pull request merged**